_posts/2021-10-04-monitor-home-internet-with-elastic.md

```diff
@@ -0,0 +1,60 @@
+
+I suspect this is a widespread issue as more people are staying home and doing things that exercise both local and global internet connections.
+
+My ISP seems to do a decent job keeping things like video streaming prioritized, but work can be a big pain. Operations like apt update/install on a local VMs, npm install, browsing company websites can get _very_ slow. It's even worse when I'm on a full-tunnel VPN as I suspect in this case my ISP can't make any sort of traffic priority decisions.
+
+For the last year, I mostly ignored it. With so much of the world suffering and dieing, it's difficult to complain about slow internet.
+
+But it's gone on so long now I decided to at least get scientific about it.
+
+## The approach
+
+I figured if I could have something like [Ookla Speedtest](https://www.speedtest.net/) running continually and storing data, I could demonstrate and quantify the issue.
+
+But what to use?
+
+### Internet Speed Logger
+
+Having recently helped out [Elastic Uptime](https://www.elastic.co/uptime-monitoring) I first reached out there.
+
+It was notably outside their intended scope, but the project's PM mentioned he'd done similar home network testing with [Brennen Smith's Internet Speed Logger](https://github.com/brennentsmith/internet-speed-logger)
+
+This got me some quick results:
+
+![screenshot of internet speed logger showing bumpy traffic graphs](/images/internet-speed-logger.png){:width="80%"}
+
+But it left something to be desired. It was hard to see what time the dips were hapening and I didn't see any options to change aggregations or do other processing. I'd probably have to hack on the code directly to draw useful conclusions.
+
+Additionally this was a mongodb+node setup on docker compose. Easy to do on my MacBook, but a little trickier to run from my Raspberry Pi 4 which was the handiest stable computer on my network.
+
+### Using the Elastic Stack
+
+I mentioned the idea in passing to one of my coworkers, [Toby Brain](https://github.com/tobio) who said he did something similar with the Ookla CLI shipping data to Elasticsearch.
+
+This sounded great for the following reasons:
+
+- I could host Elasticsearch & Kibana on [cloud.elastic.co](https://cloud.elastic.co/) and not worry about data storage availability
+- Filebeat and the Ookla CLI should be light enough to run on my Raspberry Pi
+
+#### Setting up the cluster
+
+This was easy! Just go to [cloud.elastic.co](https://cloud.elastic.co/), sign up and make a cluster.
+
+If you don't happen to work at Elastic, you'll get at 14 day free trial. If you want more, you could start a [paying plan](https://www.elastic.co/pricing/), or [consider getting a job at Elastic](https://www.elastic.co/about/careers/) ;) . Or of course you could also host your own elasticsearch and kibana on a spare machine with enough resources (but probably not a Raspberry Pi). The data set is quite small. Only about 600kb for 3 days so far.
```

Comment:
> or [consider getting a job at Elastic](https://www.elastic.co/about/careers/) ;)

👍 lol

Not sure if this space is intentional
```suggestion
If you don't happen to work at Elastic, you'll get at 14 day free trial. If you want more, you could start a [paying plan](https://www.elastic.co/pricing/), or [consider getting a job at Elastic](https://www.elastic.co/about/careers/) ;). Or of course you could also host your own elasticsearch and kibana on a spare machine with enough resources (but probably not a Raspberry Pi). The data set is quite small. Only about 600kb for 3 days so far.
```

Reply:
I was waffling on whether to add it or not. Suggestion taken :)